### PR TITLE
Add .circleci/config.yml

### PR DESCRIPTION
**Pull request by chelseatroy**
### Goal(s) of this Pull Request:

- Get started on CircleCI in our repository! This CircleCI build:

1. installs dependencies (with caching for speed)
2. Runs the unit tests
3. Runs the linter

@Dexterp37 getting integration tests to work here is possible but more involved, requiring collaboration with SRE to get a service account to reference in the build. So I'm making that a separate ticket. 

### Acceptance criteria:

In order to approve, reviewer must be able to...

- [x] When one opens a new branch, the CircleCI script should run there.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#          command: "poetry run pytest -m integration"
+
+
+# Orchestrate jobs using workflows
+# See: https://circleci.com/docs/workflows/ & https://circleci.com/docs/configuration-reference/#workflows
+workflows:
+  pre-merge-workflow: # This is the name of the workflow, feel free to change it to better match your workflow.
+    # Inside the workflow, you define the jobs you want to run.
+    jobs:
+      - pre-merge-job
```
Comment (Dexterp37) [inline .circleci/config.yml:50]:
```suggestion
      - pre-merge-job

```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      # save any changes to the cache
+      - save_cache:
+          key: v1-deps-{{ checksum "poetry.lock" }}
+          paths:
+            - env
+      - run:
+          name: "Unit tests"
+          command: "poetry run pytest"
+      - run:
+          name: "Format check"
```
Comment (Dexterp37) [inline .circleci/config.yml:37]:
```suggestion
          name: "Linting check"
```

This is actually the linting mode for ruff, see [here](https://github.com/astral-sh/ruff?tab=readme-ov-file#usage)

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          command: "poetry run ruff check"
```
Comment (Dexterp37) [inline .circleci/config.yml:38]: Would you please add the format check as well? Should be `poetry run ruff format --check` (see [here](https://github.com/astral-sh/ruff?tab=readme-ov-file#usage))

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#      - run:
```
Comment (Dexterp37) [inline .circleci/config.yml:39]: Please add a reference to the ticket for working on this :) e.g. something like "# To be addressed in XXX"

**Review by Dexterp37 — CHANGES_REQUESTED** [review]
Thanks for this work. Solid starting point, I'd love to see us enable the the formatting check (since we have that available) so that we catch these problems too before merge.

File: .circleci/config.yml (hunk 4 above)
Comment (chelseatroy) [reply .circleci/config.yml:43]: Done!

File: .circleci/config.yml (hunk 3 above)
Comment (chelseatroy) [reply .circleci/config.yml:38]: Done!

**Review by Dexterp37 — APPROVED** [review]
Perfection! :shipit:

**Pull request merged**